File: src/content/docs/topics/cs/coding-problems/sliding-window/028-find-the-index-of-the-first-occurrence-approach2.py
def str_str(haystack: str, needle: str) -> int:
    return haystack.find(needle)

assert str_str("sadbutsad", "sad") == 0
assert str_str("leetcode", "leeto") == -1
assert str_str("hello", "ll") == 2
assert str_str("a", "a") == 0
assert str_str("mississippi", "issip") == 4
assert str_str("abc", "") == 0
assert str_str("", "a") == -1
print("all tests pass")

File: src/content/docs/topics/cs/coding-problems/sliding-window/028-find-the-index-of-the-first-occurrence-approach2.ts
function assert(condition: boolean, msg: string = ''): void {
    if (!condition) throw new Error(msg || 'Assertion failed');
}

function strStr(haystack: string, needle: string): number {
    return haystack.indexOf(needle);
}

assert(strStr('sadbutsad', 'sad') === 0);
assert(strStr('leetcode', 'leeto') === -1);
assert(strStr('hello', 'll') === 2);
assert(strStr('a', 'a') === 0);
assert(strStr('mississippi', 'issip') === 4);
assert(strStr('abc', '') === 0);
assert(strStr('', 'a') === -1);
console.log('all tests pass');
